MobilityData/mobility-feed-api
# feat: add a error count to latest dataset object in gtfs-feeds

**Summary:**
Closes #364

Add validation_report field to latest_dataset in endpoints `v1/gtfs_feeds` and `v1/gtfs_feeds/{feed_id}` with content:
```
            "validation_report": {
                "total_error": (int),
                "total_warning": (int),
                "total_info": (int)
            }
``` 

**Expected behavior:** 

The  `v1/gtfs_feeds` and `v1/gtfs_feeds/{feed_id}` contain the validation report totals when a report exists, none otherwise.

**Testing tips:**
- Checkout branch 
- Generate API code
```
./scripts/api-gen.sh
```
- Reset DB(**Warning your local database content will be deleted**),
```
./scripts/docker-localdb-rebuild-data.sh --populate-db --populate-test-data
```
- Start the application
```
./scripts/api-start.sh
```
- Test endpoints and verify data is properly populated. Example curl commands:
```
curl --location 'http://localhost:8080/v1/gtfs_feeds' \
--header 'Accept: application/json' 
```
```
curl --location 'http://localhost:8080/v1/gtfs_feeds/mdb-2' \
--header 'Accept: application/json' 
```


Please make sure these boxes are checked before submitting your pull request - thanks!
